server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -365,12 +398,16 @@ private Long parseNullValue(DateFieldType fieldType) {
 
         @Override
         public DateFieldMapper build(MapperBuilderContext context) {
+            final String fullFieldName = context.buildFullName(leafName());
+            boolean hasDocValuesSkipper = isDocValuesSkipperEnabled
+                && hasDocValuesSkipper(docValues.getValue(), indexMode, indexSortConfig, fullFieldName);
             DateFieldType ft = new DateFieldType(
-                context.buildFullName(leafName()),
-                index.getValue() && indexCreatedVersion.isLegacyIndexVersion() == false,
+                fullFieldName,
+                hasDocValuesSkipper == false && index.getValue() && indexCreatedVersion.isLegacyIndexVersion() == false,
```

Comment:
This is to make sure we do not have both a sparse doc values index and an inverted index.